### PR TITLE
connectors-ci: make source-file testable in airbyte-ci

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Relates to #25053 
### Fixing `source-file`
`source-file` integration tests uses docker-compose to spin up a sftp container used for testing.
To make the container orchestration  and connection work in `airbyte-ci` we should bind python integration test to the global docker host.
The docker host should also have access to the file we want to mount of sftp, to do so we move them to the /tmp folder so that both the container under test, docker host have access to it and the docker host can mount them to the `sftp` containers.

### Fixing `source-file-secure`
source-file-secure has an implicit dependency to source-file: its docker image is based on source-file docker image in which the source-file package was installed.
But, outside of the dockerized execution, executing  `import source_file` in python in `source-file-secure` fails because our pipeline did not handle installing other connector package.
To fix this problem:
* Dependency to source-file in source-file-secure is declared explicitely inside its `setup.py`  (@evantahler we previously chatted about the benefits of explicitely variants  dependencies in `setup.py`)
* We detect dependencies to mount in the dagger pipeline by running `python setup.py egg_ingo`
* We mount the required dependencies when installing the `source-file-secure` in the test environment.
* When building the `source-file-secure` docker image we mount and install the same dependencies.

### CODE REVIEW

File: airbyte-integrations/connectors/source-file-secure/Dockerfile
Comment (alafanechere) [inline airbyte-integrations/connectors/source-file-secure/Dockerfile:8]: `source-file-secure` is now fully built from Dagger.  As `source-file` needs to be mounted at build time it was easier to accomplish it with Dagger.

File: airbyte-integrations/connectors/source-file-secure/setup.py
Comment (alafanechere) [inline airbyte-integrations/connectors/source-file-secure/setup.py:11]: Compute the absolute path to the local dependencies. In dagger pipelines we mount local dependencies to `/local_dependencies`.

File: airbyte-integrations/connectors/source-file-secure/source_file_secure/source.py
Comment (alafanechere) [inline airbyte-integrations/connectors/source-file-secure/source_file_secure/source.py:24]: This is not required anymore as the source file connector is installed as a normal python deps with pip install.

**Comment by evantahler** [comment]
While out of scope for this PR, I have the opinion that using local docker containers for testing connectors is bad for a number of reasons:
* anti-patterns like this
* slow to spin up and down containers for each test/suite
* false sense of isolation. we are in general poor at clearning up files/dbs after tests.  This is a problem masked by destroying the container which appears when moving to a hosted DB, like testing snowflake
* unobservable.  If a test fails, it is hard to connect to the DB or S3 bucked to see what went wrong

To that end, I would propose that we work to remove all test containers like this, and set up persistent test hosts for all connectors

File: airbyte-integrations/connectors/source-file-secure/Dockerfile
Comment (evantahler) [inline airbyte-integrations/connectors/source-file-secure/Dockerfile:9]: Nice.

File: airbyte-integrations/connectors/source-file-secure/setup.py
Comment (evantahler) [inline airbyte-integrations/connectors/source-file-secure/setup.py:35]: 👍

File: airbyte-integrations/connectors/source-file/setup.py
Comment (evantahler) [inline airbyte-integrations/connectors/source-file/setup.py:27]: Is this a python package to work with docker?

**Review by evantahler — APPROVED** [review]
👍 from me, mainly about explicit dependencies! 
I'd wait for the other reviews on the 🐍 code before merging

File: airbyte-integrations/connectors/source-file-secure/setup.py
Comment (bnchrch) [reply airbyte-integrations/connectors/source-file-secure/setup.py:11]: Oh dear python!

I like this solution.

My only reservations are having
1. Dagger relying on `file://` to mark relative dependencies, or pip -e as this isnt always the case ([example](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/metadata_service/orchestrator/pyproject.toml#L18
))
2. our connector setup.py's relying on knowing how dagger works under the hood.

So I want to ask a question for #1 and propose a possible idea for 2

1. I dont think we can do away with this, but how come we dont mount the relative import to the same relative location in the dagger file tree?

2. What if, in the dagger machine, we symlinked the local host path to the dagger `/local_dependencies` version? Would remove the need for having `DAGGER_BUILD` in our connectors folder.

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py
Comment (bnchrch) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:123]: This must ahve been painful, well done

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py
Comment (bnchrch) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:175]: Nit: We likely will need on for pyproject.toml at some point.

**Review by bnchrch — APPROVED** [review]
Great work here @alafanechere 

This must have been annoying to work through

I had one alternative idea, but Im not positive it would work and in the spirit of "done not perfect"

File: airbyte-integrations/connectors/source-file/setup.py
Comment (alafanechere) [reply airbyte-integrations/connectors/source-file/setup.py:27]: The source-file integration test use docker-compose without specifying it as a test dependency. I added this so that our pipeline installs it within the test environment, that does not have `docker-compose` by default.

**Comment by alafanechere** [comment]
> My only reservations are having
* Dagger relying on file:// to mark relative dependencies, or pip -e as this isnt always the case ([example](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/metadata_service/orchestrator/pyproject.toml#L18))
* our connector setup.py's relying on knowing how dagger works under the hood.

I believe there's indeed a cleaner solution I'd like to tackle later:
Have a a single function that buids a python connector container by:
* Detecting its local dependencies from pyproject.toml, requirements.txt or setup.py
* Mounting these dependencies + the connector code with path relative to the airbyte repo
* Install the dependencies according to the context (eg - install tests deps if it's the container to run tests, or install a specific cdk version if the code change has cdk change etc.) 

Then we'll have a same base container on top of which we can:
* run unit/integration tests
* build for CAT and publish

I did not take this approach right now because it was too much refacto, and as we still rely on dockerfiles for python connectors  (except source-file-secure) we still have a different container for unit/integration tests (dynamically built with dagger) and one for build and publish based on the Dockerfile.

This is definitely something we'll tackle when we'll want to remove dockerfiles for Python connectors.

**Pull request merged**
